highfidelity/hifi
# Particles don't spin with head roll, new spin + rotateWithEntity properties

Also fixes: https://highfidelity.manuscript.com/f/cases/17005/Particle-Emitter-Should-Trail-parameter-not-saving-when-exporting-importing-json

Also: allows negative emitSpeed values.

Test plan:
- Create a ParticleEffect entity in Create.  Give it a texture with a recognizable orientation, like this: http://mario.nintendo.com/assets/img/home/intro/mario-pose2.png
- Walk around it with an HMD on.  Rotate your head to the side (try to check upside down).  Check from different angles.  The particles shouldn't roll with your head as they do in master, but they should still be billboarded properly.
- In Create, in the particle tab, expand the Spin section.  Set the Particle Spin to various angles and confirm that the particles follow the angles.
- Set the Spin Start and Spin Finish properties and confirm that you can make the particles rotate over time as expected.
- Set those values back to 0 and set the Spin Spread value.  Verify that the particles each have different spins within the range.
- Set the acceleration values to all 0 and the emit speed to something positive.  Particles should shoot up out of the entity.  Make the emit speed negative and they should shoot down.  Rotate the entity and the emit direction should follow the entity's up.
- Enable Spin -> Rotate With Entity.  The particles should rotate to point in the direction of the entity's up.
- Verify that the Emitter Should Trail property works as expected.  Enable it and drag the entity around and the particles should trail behind.  Disable it and the particles should all move with the entity.
- Export a particle effect with Emitter Should Trail `false` and one with Emitter Should Trail `true`.  Import them again and confirm that the trail properties are correct.
